napari/layers/_shapes_layer/shape.py

```diff
@@ -0,0 +1,573 @@
+
+        points = self._data
+
+        centers, offsets, triangles = triangulate_edge(points,
+                                                       closed=self._closed)
+        self._edge_vertices = centers
+        self._edge_offsets = offsets
+        self._edge_triangles = triangles
+
+    def shift(self, shift):
+        """Perfroms a 2D shift on the shape
+
+        Parameters
+        ----------
+        shift : np.ndarray
+            length 2 array specifying shift of shapes.
+        """
+        shift = np.array(shift)
+
+        self._face_vertices = self._face_vertices + shift
+        self._edge_vertices = self._edge_vertices + shift
+        self._box = self._box + shift
+        self._data = self._data + shift
+
+    def scale(self, scale, center=None):
+        """Perfroms a scaling on the shape
+
+        Parameters
+        ----------
+        scale : float, list
+            scalar or list specifying rescaling of shape.
+        center : list
+            length 2 list specifying coordinate of center of scaling.
+        """
+        if isinstance(scale, (list, np.ndarray)):
+            transform = np.array([[scale[0], 0], [0, scale[1]]])
+        else:
+            transform = np.array([[scale, 0], [0, scale]])
+        if center is None:
+            self.transform(transform)
+        else:
+            self.shift(-center)
+            self.transform(transform)
+            self.shift(center)
+
+    def rotate(self, angle, center=None):
+        """Perfroms a rotation on the shape
+
+        Parameters
+        ----------
+        angle : float
+            angle specifying rotation of shape in degrees.
+        center : list
+            length 2 list specifying coordinate of center of rotation.
+        """
+        theta = np.radians(angle)
+        transform = np.array([[np.cos(theta), np.sin(theta)],
+                             [-np.sin(theta), np.cos(theta)]])
+        if center is None:
+            self.transform(transform)
+        else:
+            self.shift(-center)
+            self.transform(transform)
+            self.shift(center)
+
+    def flip(self, axis, center=None):
+        """Perfroms an vertical flip on the shape
+
+        Parameters
+        ----------
+        axis : int
+            integer specifying axis of flip. `0` flips horizontal, `1` flips
+            vertical.
+        center : list
+            length 2 list specifying coordinate of center of flip axes.
+        """
+        if axis == 0:
+            transform = np.array([[1, 0], [0, -1]])
+        elif axis == 1:
+            transform = np.array([[-1, 0], [0, 1]])
+        else:
+            raise ValueError("""Axis not recognized, must be one of "{0, 1}"
+                             """)
+        if center is None:
+            self.transform(transform)
+        else:
+            self.shift(-center)
+            self.transform(transform)
+            self.shift(-center)
+
+
+class Rectangle(Shape):
```

Comment:
I'd put these classes in separate files.

Reply:
I have done this